asv_bench/benchmarks/frame_methods.py

```diff
@@ -351,15 +351,27 @@ def time_isnull_obj(self):
 
 class Fillna:
 
-    params = ([True, False], ["pad", "bfill"])
-    param_names = ["inplace", "method"]
-
-    def setup(self, inplace, method):
+    params = (
+        [True, False],
+        ["pad", "bfill"],
+        [
+            "float64",
+            "float32",
+            "object",
+            "Int64",
+            "Float64",
+        ],
```

Comment:
int64/dt64/td64/period/categorical?

Reply:
the float and object tests were added for comparison will the masked arrays (numpy float tests use the 2d algos wheras masked arrays apply the 1d algos columnwise so we expect the numpy float tests to be faster)

not sure what to do for numpy int64.

categorical uses interpolate_2d (and pad_2d_inplace and backfill_2d_inplace) so out of scope here.

after the changes https://github.com/pandas-dev/pandas/pull/39953/commits/9024d950eabe9f85f6a0fd0f08ff28b775c451dc in response to https://github.com/pandas-dev/pandas/pull/39953#discussion_r579839269 it now makes sense to add benchmarks for dt64/td64/period, although won't be an issue if shows a performance degradation https://github.com/pandas-dev/pandas/pull/36488#discussion_r579877398